v1alpha1/meshmodel.proto

```diff
@@ -501,4 +501,83 @@ message URLRule {
 
 	// PolicyRef configures which policy this URLRule references to. 
 	string policyRef = 3;
+}
+
+// Shadow Service is a copy of an existing easemesh service. The test traffic can be scheduled to the shadow service for the test.
+message ShadowService {
+	// Name is the mesh service's name.
+	string name = 1;
+
+	// Namespace is kubernetes namespace where the service is deployed.
+    string Namespace = 2;
+
+	// MySQL configuration, optional.
+	ShadowServiceMySQL mysql  = 3;
+
+	// Kafka configuration, optional.
+	ShadowServiceKafka kafka  = 4;
+
+	// Redis configuration, optional.
+	ShadowServiceRedis redis = 5;
+
+	// RabbitMQ configuration, optional.
+	ShadowServiceRabbitMQ rabbitmq  = 6;
+
+	// ElasticSearch configuration, optional.
+	ShadowServiceElasticSearch elasticsearch  = 7;
+}
+
+
+// ShadowServiceMySQL is the mysql configuration for shadow service.
+message ShadowServiceMySQL {
+	// Hosts is the jdbc address for MySQL databases.
+	repeated string  hosts = 1;
+
+    // Username for access MySQL databases.
+	string userName = 2;
+
+    // Password for access MySQL databases.
+	string password = 3;
+}
+
+// ShadowServiceKafka is the kafka configuration for shadow service.
+message ShadowServiceKafka {
+	// Hosts is the broker server for Kafka.
+	repeated string  hosts = 1;
+}
+
+// ShadowServiceRedis is the redis configuration for shadow service.
+message ShadowServiceRedis {
+	// Host is the server addresses for Redis.
+	repeated string  hosts = 1;
+
+    // Username for access Redis server.
+	string userName = 2;
+
+    // Password for access Redis server.
+	string password = 3;
+}
+
+// ShadowServiceRabbitMQ is the rabbitmq configuration for shadow service.
+message ShadowServiceRabbitMQ {
+	// Host is the server addresses for RabbitMQ server.
+	repeated string  hosts = 1;
+
+    // Username for access RabbitMQ server.
```

Comment:
Maybe we can align the lines with a tab as before.

Reply:
updated